webui/react/src/stores/userSettings.tsx

```diff
@@ -57,6 +62,30 @@ class UserSettingsStore extends PollingStore {
     });
   }
 
+  public getAll(): Observable<Loadable<State>> {
+    return this.#settings.readOnly();
+  }
+
+  public overwrite(settings: object): void {
```

Comment:
Why do you need backend changes? You should be able to just iterate over the Map and send each setting to the API.

Reply:
That was a possibility I considered, but it would require N+1 API calls instead of just 1, and any failures have the possibility of leaving the settings in a weird partially-edited state. I thought this was the better solution.

Reply:
The extra one is to first clear out the settings. Yes, detecting the changed settings is also a possibility, but then you have a worst case of N (# new settings) + M (# old settings) calls in the (admittedly unlikely) scenario where you removed all of the (M) old settings and added N new ones, plus the additional overhead of checking the two maps against each other.

Reply:
You don't actually need to clear the settings, just unset any deleted ones. On balance I would lean toward detecting changes on the client side since it reduces the complexity of the API the API itself is more implementation-agnostic.

If you _do_ want to change the API though a better solution is to simply extend the endpoints to set and clear settings to take multiple settings at once, have `updateUserSetting` take an array of updates instead of just one.

Reply:
Yes, unsetting the deleted settings is the max N+M calls version I described above. If you think it's worth deleting the completed, working api implementation and start over on that then I guess I can do that.

Reply:
I agree. That's why I considered my options and implemented the one that I felt would result in the fastest and most reliable experience. If I didn't want to manage complexity, I would have just thrown in a loop of api calls and called it a day.

Reply:
I would say that those endpoints represent different HTTP methods.

- reset: DELETE
- update: POST/PATCH
- overwrite: PUT

However, I agree that changing the update endpoint to accept an array of settings is a good idea and would simplify some existing code. I agree that you could use that to mimic the behavior of the overwrite endpoint (and could do that without even updating it to take an array by using a loop), I just don't agree that it replaces the use case for overwrite.

Reply:
I feel that making one api call is better than two. And since making the array change or adding overwrite both require making backend changes, adding an extra endpoint is a lesser evil than making an extra call. While it's unlikely, there are more and worse failure cases when more calls are added.

Reply:
The failure cases here aren't actually different. The reset can succeed and the update fail in both the single call and two call examples. That is actually why I suggested an approach where you only update ones that need updating since then you only risk failing to change the data.

Reply:
Removed the overwrite api and rewrote update to allow for passing an array. Unfortunately the settings still get changed in the DB one at a time because I couldn't figure out how to make bun conditionally bulk add/remove rows based on input.